framework/include/base/XFEMInterface.h

```diff
@@ -47,9 +48,11 @@ class XFEMInterface : public ConsoleStreamInterface
    * Constructor
    */
   explicit
-  XFEMInterface(MooseApp & app):
+  XFEMInterface(MooseApp & app, MooseSharedPointer<FEProblem> & fe_problem):
```

Comment:
It's a little odd to pass a reference to a pointer. It's not incorrect but it's slightly confusing unless you are planning to change the pointer itself.